src/blockchain/blockchain_parameters.cpp

```diff
@@ -11,6 +11,14 @@
 
 namespace blockchain {
 
+void CheckGenesisBlockSupply(Parameters &p) noexcept {
+  CAmount genesis_supply = 0;
+  for (const auto &tx : p.genesis_block.block.vtx) {
+    genesis_supply += tx->GetValueOut();
+  }
+  assert(genesis_supply == p.initial_supply);
+}
```

Comment:
I know that the point is a bit moot as bitcoin (and therefor unit-e) is compiled with assertions enabled by default, but:

Usually using `assert` should be done in a way that compiling without the assert does not do anything which is then used in the assert only.

That is: I would have written it probably using `assert(...fancy std::accumulate() == initial_supply)` (but I know that some people are already crying their heart out about the use of a functional operator).

Reply:
The none functional alternative is to just have the function return `bool` and check it and hope for the compiler/linker which would remove it for it not being used when compiled without assertions PLUS we do not really ever compile without assertions...

Reply:
I opted for the second alternative (imperative style) as it was simpler to go there from its previous state.